src/components/v5/common/ActionsContent/partials/SocialLinksTable/hooks.tsx

```diff
@@ -0,0 +1,307 @@
+            'text-purple-400 group-hover:text-base-white': !selectedLink,
+          })}
+        >
+          <DiscordLogo size={16} />
+        </span>
+      ),
+    },
+    {
+      type: LINK_TYPE.Twitter,
+      icon: (
+        <span
+          className={clsx('mr-2', {
+            'text-base-white': selectedLink === LINK_TYPE.Twitter,
+            'text-blue-200 group-hover:text-base-white': !selectedLink,
+          })}
+        >
+          <TwitterLogo size={16} />
+        </span>
+      ),
+    },
+    {
+      type: LINK_TYPE.Telegram,
+      icon: (
+        <span
+          className={clsx('mr-2', {
+            'text-base-white': selectedLink === LINK_TYPE.Telegram,
+            'text-blue-300 group-hover:text-base-white': !selectedLink,
+          })}
+        >
+          <TelegramLogo size={16} />
+        </span>
+      ),
+    },
+    {
+      type: LINK_TYPE.Github,
+      icon: (
+        <span className="mr-2">
+          <GithubLogo size={16} />
+        </span>
+      ),
+    },
+    {
+      type: LINK_TYPE.Facebook,
+      icon: (
+        <span
+          className={clsx('mr-2', {
+            'text-base-white': selectedLink === LINK_TYPE.Facebook,
+            'text-blue-400 group-hover:text-base-white': !selectedLink,
+          })}
+        >
+          <FacebookLogo size={16} />
+        </span>
+      ),
+    },
+    {
+      type: LINK_TYPE.Instagram,
+      icon: (
+        <span
+          className={clsx('mr-2', {
+            'text-base-white': selectedLink === LINK_TYPE.Instagram,
+            'text-indigo-400 group-hover:text-base-white': !selectedLink,
+          })}
+        >
+          <InstagramLogo size={16} />
+        </span>
+      ),
+    },
+  ];
+  return socialLinksButtons;
+};
+
+export const useSocialLinks = (fieldArrayMethods, name) => {
+  const validationSchema = yup
+    .object()
+    .shape({
+      name: yup.string().defined().oneOf(Object.values(LINK_TYPE)),
+      link: yup
+        .string()
+        .url()
+        .required(formatText({ id: 'editColony.form.socialLinkError' })),
+    })
+    .defined();
+
+  type FormValues = yup.InferType<typeof validationSchema>;
+
+  const {
+    socialLinksModalToggle: [
+      isSocialLinksModalOpen,
+      { toggleOff: toggleChangeSocialLinksModalOff },
+    ],
+  } = useActionSidebarContext();
+  const values = useWatch({ name });
+  const { handleSubmit, register, formState, setValue, watch } =
+    useForm<FieldValues>({
+      resolver: yupResolver(validationSchema),
+      defaultValues: {
+        name: '',
+        link: '',
+      },
+      mode: 'onChange',
+    });
+  const { field } = useController({
+    name: 'name',
+  });
+  const linkName: LinkType = watch('name');
+  const selectedLinkDetails = values?.find((link) => link.name === linkName);
+
+  const selectedFieldIndex: number = values?.findIndex(
+    (item) => item.name === linkName,
+  );
+
+  useEffect(() => {
```

Comment:
Can you explain what is going on here? I don't understand this logic

Reply:
for example when user created two links and want to edit one of them, the modal will be open and the correct link type will be marked and input should be filled by correct link